exercises/t_jubb/ex01_basics/src/find_words.cpp

```diff
@@ -0,0 +1,189 @@
+#include <iostream>
+#include <string>
+#include <vector>
+#include <fstream>
+#include <map>
+#include <set>
+#include <iomanip>
+#include <algorithm>
+#include <functional>
+
+using std::cout;
+using std::endl;
+
+
+std::vector<std::string> get_words(const std::string input_string, const int min_len) {
+  /*
+    Splits input_string up into its components words by finding a character, and then
+    iterating through characters until a non-alphabetic character is found.
+
+    @param input_string a string with spaces.
+    @param min_len, minimum lenth of word to store.
+    @return a vector of strings (the words).
+  */
+  std::vector<std::string> words;
+  
+  int index      = -1;
+  int word_start = 0;
+  int word_end   = 0;
+  int word_len   = 0;
+
+  for ( std::string::const_iterator i = input_string.begin(); i <= input_string.end(); i++ ) {
+    index++;
+    // If we find the start of a word.
+    if (isalpha(*i) != 0){
+      word_start = index;
+      // Iterate through the string until the character is not part of a word.
+      while ((isalpha(*i) != 0)) {
+        index++;
+        i++;
+        // Catch the end of the string for last word
+        if ((i == input_string.end())) {
+          break;
+        }
+      }
+      word_end = index;
+      word_len = word_end - word_start;
+      if ( word_len > min_len){
+        words.push_back(input_string.substr(word_start,(word_end-word_start)));
+      }
+    }
+  }
+
+  return words;
+};
+
+int add_word_counts(std::map<std::string, int>& current_counts , const std::vector<std::string>& word_vec  ) {
+  /*
+    Adds a vector of words into a map of <word,occurance> pairs.
+
+    @param current_counts, a map of <word,occurance> pairs.
+    @param word_vec a vector of words.
+    @return current_counts after the words in word_vec have been added.
+  */
+  for ( std::vector<std::string>::const_iterator i = word_vec.begin(); i != word_vec.end(); i++ ) {
+      ++current_counts[(*i)];
+  };
+  // return current_counts;
+  return 0;
+};
+
+int ordered_by_value(std::vector< std::pair<  int , std::string > >& vec ,std::map<std::string, int >& count_map) {
+  /*
+    Sorts a map of <string,value> pairs by value (ascending), by inserting the <value,string> pairs into a vector
+    and then performing the default sort.
+
+    @param vec : a vector to append the entries of count_map to.
+    @param count_map : a map of <string,value> pairs.
+
+  */
+  std::pair<int,std::string> elem;
```

Comment:
I think this would be a good use of [`std::transform`](http://en.cppreference.com/w/cpp/algorithm/transform) and a lambda:
```c++
  std::vector<std::pair<int, std::string>> ordered;
  ordered.resize(count_map.size()); // transform doesn't resize the output range
  std::transform(std::begin(count_map), std::end(count_map), std::begin(ordered), 
    [](const std::pair<std::string, int> & src_elem) { return std::make_pair(src_elem.second, src_elem.first); });
  std::sort(std::begin(ordered), std::end(ordered));
```